plugins/steve/resource-class.js

```diff
@@ -680,7 +680,7 @@ export default class Resource {
   }
 
   get canYaml() {
-    return this.hasLink('view');
+    return this.hasLink('view') && !this.schema?.resourceMethods.find(x => x === 'blocked-PUT'); // 'blocked-PUT' will prevent config from saving when editing as yaml
```

Comment:
I suggested this, `canYaml` is used for the edit yaml / clone actions and also the list masthead.

Though looking at it again it also covers cases where only view is needed. Think there need a new `canEditYaml` getter and some usages of `canYaml` switched over

Reply:
I've moved the logic that checks for `blocked-PUT` to a getter `canEditYaml`. Now instead of asking `canUpdate` to determine Editing or Viewing of the yaml, it checks for `blocked-PUT` first.

Reply:
It looks like there's one last place canYaml is being used control the display of a 'create yaml' button: https://github.com/rancher/dashboard/blob/master/components/ResourceList/Masthead.vue#L101
Once that is converted to canEditYaml I think the PR is good to go.

Reply:
Converted.